File: Dante1/Module1.02/Zadanie2.04/main.c
#include <stdlib.h>
#include <stdio.h>

int main()
{
    int num1,num2;
    char oper;

    printf("Enter something: ");
    int check = scanf("%d%ci%d",&num1,&oper,&num2);
    if(check != 3 || num2 < 0)
    {
        printf("Incorrect input");
        return 1;
    }

    if(oper == '-')
    {
        printf("Real: %i\n",num1);
        printf("Image: %i\n",-1 * num2);
    }
    else if(oper == '+')
    {
        printf("Real: %i\n",num1);
        printf("Image: %i\n",num2);
    }
    else
    {
        printf("Incorrect input");
        return 1;
    }
    return 0;
}

File: Dante1/Module1.02/Zadanie2.04/main.rs
use regex::Regex;
fn main()
{
    println!("Enter: ");

    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();

    let input = input.trim();

    let re = Regex::new(r"^([+-]?\d+)\s*([+-])i(\d+)$").unwrap();

    if let Some(caps) = re.captures(input)
    {
        let real: i32 = caps[1].parse().unwrap();
        let sign = &caps[2];
        let imag: i32 = caps[3].parse().unwrap();

        let imag = 
            if sign == "-"
            { 
                -imag 
            }
            else 
            { 
                imag 
            };

        println!("Real part: {}", real);
        println!("Imaginary part: {}", imag);
    }
    else
    {
        println!("Incorrect input");
    }
}
